### DISCUSSION TITLE
Allure reports not generate result of all test files

### DISCUSSION THREAD

**Discussion by mariamaslam** [mmisty/cypress-allure-adapter#55] [Q&A] [answered]
hi @mmisty 
I'm facing an issue with allure report i have around 42 test file and in the reports only 23 are visible and in my script, there is nothing mentioned to run only specific scripts. 

This is package.json

 ```
 "scripts": {
    "cy:run": "cypress run --config specPattern=cypress/e2e*.js --env allure=true || true && allure generate allure-results --clean -o allure-report && allure open allure-report",
    "allure:report": "npm run allure generate allure-result && allure open allure-result",
    "beforetest": "npm run clean:folders",
    "tests": "npm run cy:run || npm run aftertest",
    "aftertest": "npm run allure:report"
```
this is config.js

```
const { defineConfig } = require("cypress");
const { configureAllureAdapterPlugins } = require("@mmisty/cypress-allure-adapter/plugins");

module.exports = defineConfig({
  e2e: {
    baseUrl: "....,
    env: {
      allure: true,
      allureCleanResults: true,
      allureSkipCommands: 'wrap',
      allureResults: 'allure-results',
      // when using Allure TestOps:
      // allureResultsWatchPath: 'allure-results/watch'
    },
    responseTimeout: 120000,
    defaultCommandTimeout: 15000,
    video: true,
    
    
    setupNodeEvents(on, config) {
      config.fixturesFolder = "cypress/fixtures"; 
      config.specPattern = [
        'cypress/e2e/Accounts/1.cy.js',
        'cypress/e2e/Accounts/2.cy.js',
        'cypress/e2e/Accounts/3cy.js',
       'cypress/e2e/Accounts/4o.cy.js',
       'cypress/e2e/Accounts/5.cy.js',
        'cypress/e2e/Accounts/6.cy.js',
         'cypress/e2e/Accounts/7.cy.js',
         'cypress/e2e/Accounts/8.cy.js',
         'cypress/e2e/Accounts/9.cy.js',
         'cypress/e2e/Accounts/10.cy.js',
         'cypress/e2e/Accounts/11.cy.js',
         'cypress/e2e/Accounts/12.cy.js',
         'cypress/e2e/Accounts/13.cy.js',
         'cypress/e2e/Accounts/14.cy.js',
         'cypress/e2e/Accounts/15.cy.js',
         'cypress/e2e/Accounts/16.cy.js',
         'cypress/e2e/Accounts/17.cy.js',
         'cypress/e2e/Accounts/18.cy.js',
         'cypress/e2e/Accounts/19.cy.js',
         'cypress/e2e/Orders/1.cy.js',
         'cypress/e2e/Orders/2.cy.js',
         'cypress/e2e/Orders/3.cy.js',
         'cypress/e2e/Orders/4.cy.js',
         'cypress/e2e/Orders/5.cy.js',
         'cypress/e2e/Orders/6.cy.js',
         'cypress/e2e/Orders/7.cy.js',
         'cypress/e2e/Orders/8.cy.js',
         'cypress/e2e/Orders/9.cy.js',
         'cypress/e2e/Orders/10.cy.js',
         'cypress/e2e/Orders/11.cy.js',
         'cypress/e2e/Orders/12.cy.js',
         'cypress/e2e/Orders/13.cy.js',
         'cypress/e2e/Orders/14.cy.js',
         'cypress/e2e/Orders/15.cy.js',
         'cypress/e2e/Orders/16.cy.js',
         'cypress/e2e/Orders/17.cy.js',
         'cypress/e2e/Orders/18.cy.js',
         'cypress/e2e/Orders/19.cy.js',
         'cypress/e2e/Orders/20.cy.js',
         'cypress/e2e/Orders/21.cy.js',
         'cypress/e2e/Orders/22.cy.js',
         'cypress/e2e/Orders/23.cy.js',
  
  
      ]

     // exclude in case you want to exclude any tests
     config.excludeSpec=[
      'cypress/e2e/Accounts/EmploymentTypeOthers.cy.js', //this is still under testing'
    ]
      const reporter = configureAllureAdapterPlugins(on, config);
      //require("cypress-localstorage-commands/plugin")(on, config);

      console.log(' === ENVIRONMENT:');
      console.log(config.env);
      console.log(' === ');
  
      // this is to write categories and environment information
      on('before:run', details => {
        reporter?.writeEnvironmentInfo({
          info: {
            os: details.system.osName,
            osVersion: details.system.osVersion,
            browser: details.browser?.displayName + ' ' + details.browser?.version,
            ...config.env
          },
        }); 
  
         reporter?.writeCategoriesDefinitions({ categories: './allure-error-categories.json' });
      });
      
      

      return config;
    },
  },
});
```

I have renamed the APIS to 1 2 because of obvious reason :)

**Comment by mmisty** [comment]
@mariamaslam Hello, 

Could you please attach your report and show the contents of a any test file

**Comment by mmisty** [comment, marked as answer]
@mariamaslam regarding your configuration
1. script `"allure:report": "npm run allure generate allure-result && allure open allure-result",` seems incorrect for me.
Should be `"allure:report": "npm run allure -- generate allure-result && allure open allure-result",`
2. you've setup plugin to write results into `allure-results` folder, but you generate report for `allure -result` folder

**Reply by mariamaslam** [reply]
Sure, let me try this right away n get back to you.

**Reply by mariamaslam** [reply]
@mmisty now my reports are Nan
I changed this "allure:report": "npm run allure -- generate allure-result && allure open allure-result",
update the folder name is config.js

**Reply by mmisty** [reply]
@mariamaslam oh, sorry, 
this is the correct command (and it is btw not related to this plugin) 
`"allure:report": "allure generate allure-result && allure open allure-result"`

**Reply by brian-gn** [reply]
Hey there, thank you so much for this contribution! I'd like to throw my 2 cents regarding the correct report command (yours was fine except it was missing the "s" in "allure-results", which would be: `"allure:report": "allure generate allure-results --clean -o allure-report"`

**Comment by mariamaslam** [comment]
Yes, here is the screenshot attached

```
/// <reference types="Cypress" />
describe(' Protection', () => {

    it('Verify user is able to enable/disable their shield', () => {
      cy.accessTokenCards();
      cy.readFile('cypress/fixtures/Data.json').then(({ globalProspectId }) => {
        const token = Cypress.env('token'); // Retrieve the access token{
  
  
      cy.api({
        failOnStatusCode: false,
        method: 'PATCH',
        url: `//prospects/${globalProspectId}`,
        headers: {
          "Authorization": `Bearer ${token}`
        },
        body: {
            "creditShieldProtectDetails": {
                "creditShieldReqd": "N"
              },
              "language": "EN"
    }
      }).then((response) => {
        // Assert that the OTP was generated successfully
        expect(response.status).to.eq(200);
  
  

      });
    });
  });
});
```

<img width="929" alt="Screenshot 2023-10-30 at 11 27 20 AM" src="https://github.com/mmisty/cypress-allure-adapter/assets/47422016/468d3f9c-7512-483d-8a3d-7a8b63a1b71a">

<img width="723" alt="Screenshot 2023-10-30 at 11 26 13 AM" src="https://github.com/mmisty/cypress-allure-adapter/assets/47422016/2729bbc3-ba6e-457a-9d19-a6735896abf0">

**Reply by mmisty** [reply]
You have a configuration issue, and to solve that please check: 
 - that after running tests you have `allure-results` folder with `*-result.json` files in it
 - that you generate report for this folder (not any other) 
 
And also could you please read this section https://github.com/mmisty/cypress-allure-adapter/blob/main/README.pack.md#support

**Reply by mariamaslam** [reply]
Im sorry, but im not getting you, what needs to be done exactly is my issue in config.js or package.json? @mmisty

**Reply by mariamaslam** [reply]
my reports are NaN  after making these changes :'(

I have a critical demo in 4 hours from now :(
Please support @mmisty

**Reply by mmisty** [reply]
@mariamaslam I don't know what needs to be done exactly since It is your project and I don't see it. 
I referred you what to check within your project.


What I think is happening that you generate report for different folder

please do the following
1. delete `allure-results` and `allure-result` folders
2. check that env variable `allureResults` is set to `allure-results`
3. run tests: `cypress run`
4.  check `allure-results` folder has `*-result.json` files
5.  serve report: `allure serve`
6. if previous step is successful and you see what you expected to see generate report by `allure generate` and open it with `allure open`

If some step doesn't work for you please check everything once again.

Also it worth reading
 - cypress documentation (command line options  - for example specPattern )
 - npm documentation 
 - allure documentation (how to generate report)

**Reply by mariamaslam** [reply]
ok thanks will look into it.

Please can you confirm regarding any configuration that can help in emailing the report or generating a pdf?

**Reply by mmisty** [reply]
@mariamaslam 
you can try `allure generate --single-file` - should generate single file html.
To create PDF you need to build your own html from results (you can try using allure-js-parser to get parsed results into an object)
